[example] adde a groth16 example (#21329)
MystenLabs/sui: examples/groth16/README.md

# Sui Groth16 Example

> This is a simple example to the Sui Groth16 documentation.

Reviewer: ```suggestion
This is an example to the Sui Groth16 documentation.
```
Avoid using the term 'simple' in tech writing. If it's not simple for whatever reason to the user, you're essentially calling them dumb.
